roxctl/scanner/downloaddb/downloaddb.go

```diff
@@ -0,0 +1,328 @@
+	} else {
+		versionVariants := disectVersion(version)
+		for _, versionVariant := range versionVariants {
+			bundleFileNames = append(bundleFileNames, fmt.Sprintf(bundleFileNameFmt, versionVariant))
+		}
+	}
+
+	return bundleFileNames, nil
+}
+
+// detectVersion attempts to determine an appropriate base version to use.
+func (cmd *scannerDownloadDBCommand) detectVersion() string {
+	if cmd.version != "" {
+		cmd.env.Logger().InfofLn("Using version from command line: %q", cmd.version)
+		return cmd.version
+	}
+
+	if !cmd.skipCentral {
+		if ver, err := cmd.versionFromCentral(); err == nil {
+			cmd.env.Logger().InfofLn("Using version from Central: %q", ver)
+			return ver
+		}
+	}
+
+	ver := version.GetMainVersion()
+	cmd.env.Logger().InfofLn("Using version from roxctl binary: %q", ver)
+	return ver
+}
+
+// versionFromCentral attempts to pull version from Central's metadata
+// service.
+func (cmd *scannerDownloadDBCommand) versionFromCentral() (string, error) {
+	client, err := cmd.env.HTTPClient(5 * time.Second)
+	if err != nil {
+		cmd.env.Logger().WarnfLn("issue building central http client: %v", err)
+		return "", err
+	}
+
+	resp, err := client.DoReqAndVerifyStatusCode("v1/metadata", http.MethodGet, http.StatusOK, nil)
+	if err != nil {
+		cmd.env.Logger().WarnfLn("error contacting central: %v", err)
+		return "", err
+	}
+	defer utils.IgnoreError(resp.Body.Close)
+
+	var metadata v1.Metadata
+	if err := jsonpb.Unmarshal(resp.Body, &metadata); err != nil {
+		cmd.env.Logger().WarnfLn("error reading metadata from central: %v", err)
+		return "", err
+	}
+
+	if err := resp.Body.Close(); err != nil {
+		cmd.env.Logger().WarnfLn("could not close response body: %w", err)
+		return "", err
+	}
+
+	return metadata.GetVersion(), nil
+}
+
+// buildAndValidateOutputFileName returns a validated output file name where
+// downloaded data should be stored.
+func (cmd *scannerDownloadDBCommand) buildAndValidateOutputFileName(bundleFileName string) (string, error) {
+	outFileName := bundleFileName
+
+	if cmd.filename != "" {
+		outFileName = cmd.filename
+
+		if cmd.filenameValidated {
+			return outFileName, nil
+		}
+		cmd.filenameValidated = true
+	}
+
+	if !cmd.force {
+		// Throw an error if the file exists and force flag not set.
+		if _, err := os.Stat(outFileName); err == nil {
+			return "", fmt.Errorf("file %q already exists, to overwrite use `--force`", outFileName)
+		}
+	}
+
+	return outFileName, nil
+}
+
+// buildDownloadURL returns the URL from which to download the vulnerability
+// database from.
+func (cmd *scannerDownloadDBCommand) buildDownloadURL(bundleFileName string) (string, error) {
+	return url.JoinPath(env.ScannerDBDownloadBaseURL.Setting(), bundleFileName)
+}
+
+// downloadVulnDB downloads the vulnerability database from url and stores it in
+// the provided output file.
+func (cmd *scannerDownloadDBCommand) downloadVulnDB(url string, outFileName string) error {
+	resp, err := http.Get(url)
```

Comment:
This is a bit unfortunate that we have to use another client and it won't do things like retries. For simplicity, WDYT about re-using the `retryableHTTP` one here?

Reply:
https://github.com/stackrox/stackrox/pull/9803/commits/44b99851cdf0edbf6648c2bf7b49e0ecaa6da23d